IRCT-RI/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/ri/i2b2/I2B2XMLResourceImplementation.java

```diff
@@ -855,40 +855,57 @@ private void convertPatientDataResponseTypeToResultSet(PatientDataResponseType p
 		// if not, just put concept_cd as the column name
 		// So we need a alias name and concept_cd mapping as well for later append rows....
 		Map<String, String> conceptCD_aliasName_Map = new HashMap<>();
+
+		// conceptPath, the format of conceptPath is \xxx\xxxx\xxxxxx\
+		// but format of the key in aliasMap is \\domainname\xxx\xxxx\xxxxx\
+		// we need to pre-process the selectMap to make key match the format of conceptPath
+		Map<String, String> preProcessedSelectMap = new LinkedHashMap<>();
+
+		for (Map.Entry<String, String> entry : selectMap.entrySet()){
+			String preProcessedKey = "\\";
+			String[] splitKeys = entry.getKey().split("\\\\");
```

Comment:
Are you splitting it on one \ or two?  The notes above suggest there's only one \ except in the beginning.

Reply:
Maybe it needs to be escaped?

Reply:
Right, there's two and each is escaped so there's four total. I was just confused and frightened by so many \s.

Reply:
Haha. Yes. First time when I saw this, I was frightened as well and tested again and again to make sure it's correct. The thing is the parameter in function split() is expecting a regex. When things go to regex, it will be doubled. So to split one '\', first escape to as "\\" then go to regex, each '\' will need to be escaped. In the end, it turns out "\\\\". 

It will help you to understand more if you check the source code of String.split(regex)